### PR TITLE
WIP [Orders] add wait for order state transition

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/personal_data/orders/order_state.py [hunk 1]
```diff
@@ -146,7 +146,8 @@ async def _refresh_order_from_exchange(self, force_synchronization=False) -> boo
                                          self.order.exchange_manager.id).get_internal_producer().
                 update_order_from_exchange(order=self.order,
                                            wait_for_refresh=True,
-                                           force_job_execution=force_synchronization))
+                                           force_job_execution=force_synchronization,
+                                           wait_for_order_state_transition=True))
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/order_state.py:150]: Why not using `wait_for_refresh` ?

Comment (GuillaumeDSM) [reply octobot_trading/personal_data/orders/order_state.py:150]: We can but it's an AsyncJob parameter that is not passed to callbacks for now (only **kwargs are), I don't think it's a good idea to pass it to every callbacks.

**Comment by Herklos** [comment]
Should we close this @GuillaumeDSM?

**Comment by GuillaumeDSM** [comment]
Closed till we have a real need for this

**Pull request closed without merging**